server/src/main/java/org/opensearch/common/util/FeatureFlags.java

```diff
@@ -93,6 +99,11 @@ public static boolean isEnabled(String featureFlagName) {
 
     public static final Setting<Boolean> REMOTE_STORE_SETTING = Setting.boolSetting(REMOTE_STORE, false, Property.NodeScope);
 
+    public static final Setting<Boolean> SEGMENT_REPLICATION_WITH_REMOTE_STORE_EXPERIMENTAL_SETTING = Setting.boolSetting(
```

Comment:
I think we can reuse the existing SEGMENT_REPLICATION_EXPERIMENTAL flag here, treating it as an experimental mode for segrep that would include multiple features.  Given we can run segrep + remote store side by side I believe we need an additional index level setting to determine `isSegRepWithRemoteEnabled`.